src/server/endpoints/covidcast.py

```diff
@@ -170,3 +173,78 @@ def gen(rows):
 
     # now use a generator for sending the rows and execute all the other queries
     return p(gen(r))
+
+
+@bp.route("/csv", methods=("GET", "POST"))
+def handle_export():
+    source, signal = request.args.get("signal", "jhu-csse:confirmed_incidence_num").split(":")
+    start_day = request.args.get("start_day", "2020-04-01")
+    end_day = request.args.get("end_day", "2020-09-01")
+    geo_type = request.args.get("geo_type", "county")
+    geo_values = request.args.get("geo_values", "*")
+
+    if geo_values != "*":
+        geo_values = geo_values.split(",")
+
+    as_of = request.args.get("as_of", None)
+
+    start_day = datetime.strptime(start_day, "%Y-%m-%d").date()
+    end_day = datetime.strptime(end_day, "%Y-%m-%d").date()
+
+    if as_of is not None:
+        as_of = datetime.strptime(as_of, "%Y-%m-%d").date()
+
+    # build query
+    q = QueryBuilder("covidcast", "t")
+
+    q.set_fields(["geo_value", "signal", "time_value", "issue", "lag", "value", "stderr", "sample_size", "geo_type", "source"], [], [])
+    q.set_order("geo_value", "time_value")
+    q.where(source=source, signal=signal, time_type="day")
+    q.conditions.append("time_value BETWEEN :start_day AND :end_day")
+    q.params["start_day"] = date_to_time_value(start_day)
+    q.params["end_day"] = date_to_time_value(end_day)
+    q.where_geo_pairs("geo_type", "geo_value", [GeoPair(geo_type, True if geo_values == "*" else geo_values)])
+
+    _handle_lag_issues_as_of(q, None, None, date_to_time_value(as_of) if as_of is not None else None)
+
+    # tag as_of in filename, if it was specified
+    as_of_str = "-asof-{as_of}".format(as_of=as_of.isoformat()) if as_of is not None else ""
+    filename = "covidcast-{source}-{signal}-{start_day}-to-{end_day}{as_of}".format(source=source, signal=signal, start_day=start_day.isoformat(), end_day=end_day.isoformat(), as_of=as_of_str)
+    p = CSVPrinter(filename)
```

Comment:
This is a good idea but we should make sure the covidcast library isn't expecting a universal/static filename before merging

Reply:
I don't understand, this new endpoint is solely for  https://delphi.cmu.edu/covidcast/export/ to replace the covidcast-csv-server

Reply:
ah, I misunderstood. This is fine then